src/entt/entity/fwd.hpp

```diff
@@ -28,7 +28,10 @@ class basic_sparse_set;
 template<typename Type, typename = entity, typename = std::allocator<Type>, typename = void>
 class basic_storage;
 
-template<typename Type>
+template<typename Entity = entity, typename = std::allocator<Entity>>
+class basic_registry;
+
+template<typename Storage, typename Registry = basic_registry<typename Storage::entity_type, typename Storage::base_type::allocator_type>>
```

Comment:
Not sure about this one in the long term but let it be, we can still change it later on. 👍